src/_P118_Itho.ino

```diff
@@ -244,134 +246,149 @@ boolean Plugin_118(byte function, struct EventStruct *event, String &string)
 	case PLUGIN_WRITE: {
 		String tmpString = string;
 		String cmd = parseString(tmpString, 1);
-		String param1 = parseString(tmpString, 2);
 			if (cmd.equalsIgnoreCase(F("STATE")))
 			{
-
-				if (param1.equalsIgnoreCase(F("1111")))
-				{
-					noInterrupts();
-					PLUGIN_118_rf.sendCommand(IthoJoin);
-					interrupts();
-					PLUGIN_118_rf.initReceive();
-					addLog(LOG_LEVEL_INFO, F("Sent command for 'join' to Itho unit"));
-					printWebString += F("Sent command for 'join' to Itho unit");
-					success = true;
-				}
-				if (param1.equalsIgnoreCase(F("9999")))
-				{
-					noInterrupts();
-					PLUGIN_118_rf.sendCommand(IthoLeave);
-					interrupts();
-					PLUGIN_118_rf.initReceive();
-					addLog(LOG_LEVEL_INFO, F("Sent command for 'leave' to Itho unit"));
-					printWebString += F("Sent command for 'leave' to Itho unit");
-					success = true;
-				}
-			  if (param1.equalsIgnoreCase(F("0")))
+				switch(event->Par1) {
+					case 1111:
+					{
+						noInterrupts();
+						PLUGIN_118_rf.sendCommand(IthoJoin);
+						interrupts();
+						PLUGIN_118_rf.initReceive();
+						addLog(LOG_LEVEL_INFO, F("Sent command for 'join' to Itho unit"));
+						printWebString += F("Sent command for 'join' to Itho unit");
+						success = true;
+						break;
+					}
+				 	case 9999:
+					{
+						noInterrupts();
+						PLUGIN_118_rf.sendCommand(IthoLeave);
```

Comment:
Good question if it is needed, I added this ~2 years ago because the plugin kept crashing. The source of this could have been somewhere else, though. The philosophy is that when we send we don't want an arriving RF packet to interrupt the sending, otherwise the actual state and send state might not be the same. The reason it is here and not in the library is that others also use the library in their projects and this makes it easier for me to update the library when changes are made.

I can disable the noInterrupts on my side and see if I notice any issues.

Reply:
You could also add a `sendCommandNoInterrupt()` to the library that does just that, disable interrupt, send the command and re-enable the interrupt, killing 2 birds with 1 stone 😉

Reply:
And since it is called several times, it will probably also reduce the binary size.

Reply:
Nope, not a good idea, since you're writing to the log too, which may take "forever" and is very tricky to do without interrupts enabled.

Also "initreceive" was outside the scope of the disabled interrupts, so no idea what happens when that's initialized with interrupts disabled.

Disabling interrupts should be as short as possible (and not shorter ;)) and above all not longer than needed to keep the code more clean.
Better to find some way, like Ton mentioned, to have a function handling the interrupts and sending commands.

Reply:
I have moved the blocking code to the Itho library and also reduced it in the ITHOCheck routine that handles received RF packages.